packages/legend-graph/src/models/protocols/pure/v1/transformation/pureProtocol/serializationHelpers/V1_RuntimeSerializationHelper.ts

```diff
@@ -71,40 +73,44 @@ export const V1_legacyRuntimeModelSchema = createModelSchema(V1_LegacyRuntime, {
   ),
 });
 
-export const V1_identifiedConnectionModelSchema = createModelSchema(
-  V1_IdentifiedConnection,
-  {
+export const V1_setupEngineRuntimeModelSchema = (
+  plugins: PureProtocolProcessorPlugin[],
+): void => {
+  createModelSchema(V1_IdentifiedConnection, {
     connection: custom(
-      (val) => V1_serializeConnectionValue(val, true),
-      (val) => V1_deserializeConnectionValue(val, true),
+      (val) => V1_serializeConnectionValue(val, true, plugins),
+      (val) => V1_deserializeConnectionValue(val, true, plugins),
     ),
     id: primitive(),
-  },
-);
-
-export const V1_storeConnectionModelSchema = createModelSchema(
-  V1_StoreConnections,
-  {
+  });
+  createModelSchema(V1_StoreConnections, {
     store: usingModelSchema(V1_packageableElementPointerDeserrializerSchema),
     storeConnections: list(
-      usingModelSchema(V1_identifiedConnectionModelSchema),
+      custom(
+        (val) => serialize(V1_IdentifiedConnection, val),
+        (val) => deserialize(V1_IdentifiedConnection, val),
+      ),
     ),
-  },
-);
-
-export const V1_engineRuntimeModelSchema = createModelSchema(V1_EngineRuntime, {
-  _type: usingConstantValueSchema(V1_RuntimeType.ENGINE_RUNTIME),
-  connections: list(usingModelSchema(V1_storeConnectionModelSchema)),
-  mappings: list(
-    usingModelSchema(V1_packageableElementPointerDeserrializerSchema),
-  ),
-});
+  });
+  createModelSchema(V1_EngineRuntime, {
+    _type: usingConstantValueSchema(V1_RuntimeType.ENGINE_RUNTIME),
+    connections: list(
+      custom(
+        (val) => serialize(V1_StoreConnections, val),
```

Comment:
Like the above, you can do `list(object(V1_StoreConnections))`

Reply:
I used the same initially. It was popping me an error saying: expected prop schema as first argument.

Reply:
ok, that's weird. I, unfortunately, don't have time to check this out now, I think for now you can do
`list(usingModelSchema(V1_storeConnectionsModelSchema))` and ditto for the ones below

Reply:
Do you want me to add methods for v1_storeconnectionsModelschema, v1_identifiedConnectionModelSchema ?

Reply:
If I do that then I have pass plugins as a parameter to those methods.

Reply:
no :) not method, but in the body of `V1_setupEngineRuntimeSerialization()`, assign the return value for `createModelSchema` to those variables

Reply:
Then I won't be able to pass V1_engineRuntimeModelSchema inside v1_packageableRuntimeModelSchema because i cannot access it outside V1_setupEngineRuntimeSerialization()

Reply:
oh for that particular one, keep using `custom()` is fine